examples/generative/diffusion/generate.py

```diff
@@ -0,0 +1,334 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pickle  # TODO remove
+
+import hydra
+import numpy as np
+import PIL.Image
+import torch
+import tqdm
+from omegaconf import DictConfig
+from utils import StackedRandomGenerator, open_url
+
+from modulus.distributed import DistributedManager
+from modulus.launch.logging import PythonLogger, RankZeroLoggingWrapper
+
+
+def sampler(
```

Comment:
This sampler is overly complex, poorly documented, and not modular. If we intend to support generative models like these in the future, it would behoove us to have general purpose langevin samplers.

Reply:
Added to this epic: https://github.com/NVIDIA/modulus/issues/196